common/djangoapps/track/segment.py

```diff
@@ -11,15 +11,31 @@
 from urlparse import urlunsplit
 
 import analytics
+from crum import get_current_request
 from django.conf import settings
 from eventtracking import tracker
+from track import contexts
+from track.views import _get_request_value
 
 
-def track(user_id, event_name, properties=None, context=None):
+def track(user_id, event_name, properties=None, context=None, send_to_track=False, user=None):
     """Wrapper for emitting Segment track event, including augmenting context information from middleware."""
+    properties = properties or {}
+
+    if event_name is not None and send_to_track:
```

Comment:
nit:
```suggestion
    if event_name and send_to_track:
```

Reply:
This is how edX has used it. Also, I know it won't happen but, if someone sends `''` as even name, that will get rejected too. So let's leave it as it is?

Reply:
if there is no event name, will it be caliper compliant?

Reply:
Not sure but could be. Event name is just used in the extensions when we transform it into caliper format.